src/NuGet.Core/NuGet.Protocol/Converters/SemanticVersionConverter.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// A SemanticVersion JSON converter.
+    /// </summary>
+    public sealed class SemanticVersionConverter : JsonConverter
+    {
+        /// <summary>
+        /// Gets a flag indicating whether or not a type is convertible.
+        /// </summary>
+        /// <param name="objectType">An object type to check.</param>
+        /// <returns><c>true</c> if <paramref name="objectType" /> is convertible; otherwise <c>false</c>.</returns>
+        public override bool CanConvert(Type objectType) => objectType == typeof(SemanticVersion);
+
+        /// <summary>
+        /// Reads the JSON representation of the object.
+        /// </summary>
+        /// <param name="reader">A JSON reader.</param>
+        /// <param name="objectType">The type of the object.</param>
+        /// <param name="existingValue">The existing value of the object.</param>
+        /// <param name="serializer">A serializer.</param>
+        /// <returns></returns>
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            return reader.TokenType != JsonToken.Null ? SemanticVersion.Parse(serializer.Deserialize<string>(reader)) : null;
+        }
+
+        /// <summary>
+        /// Writes the JSON representation of the object.
+        /// </summary>
+        /// <param name="writer">A JSON writer.</param>
+        /// <param name="value">A value to serialize.</param>
+        /// <param name="serializer">A serializer.</param>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var semanticVersion = (SemanticVersion)value;
```

Comment:
From [Json.NET documentation](http://www.newtonsoft.com/json/help/html/SerializationSettings.htm#Converters):

>When a JsonConverter has been added to a JsonSerializer it will be checked for every value that is being serialized/deserialized using its CanConvert to see if it should be used. If CanConvert returns true then the JsonConverter will be used to read or write the JSON for that value.

This converter only supports the [`SemanticVersion` type](https://github.com/NuGet/NuGet.Client/pull/1328/files#diff-08ad61993913bfd0eb74363e2afddc80R20), so if the above JsonConverter contract changes or if another value type is passed into `WriteJson(...)` that's an error condition.